insights/parsers/satellite_installer_configurations.py

```diff
@@ -25,5 +26,8 @@ class CustomHiera(YAMLParser):
         >>> custom_hiera['apache::mod::prefork::serverlimit']
         582
     """
-
-    pass
+    def parse_content(self, content):
+        try:
+            super(CustomHiera, self).parse_content(content)
+        except ParseException:
+            pass
```

Comment:
@huali027 I believe this could results in problems for rules if the Yaml has a bad format.  If the Yaml parser fails with an `ScannerException` instead of parsing a file that simply did not contain anything but comments (which causes a `SkipException`), this code will still pass and the `data` attribute will not be present in the resulting parser object because it was never initialized.  If you add a test with invalid yaml, your test will fail.  I think you'd actually want to fail with the `ParseException` in that case.

Reply:
@bfahr yes, @xiangce has pointed it out. So I updated the parent class to raise a SkipException when the file is empty since it is not parsing error. Since the PR for parent class hasn't been merged, so I didn't modify this PR yesterday. I updated it now, please have a check again. Thanks for your review.